detox/src/devices/common/drivers/android/tools/apk.js

```diff
@@ -0,0 +1,29 @@
+const path = require('path');
+
+const _ = require('lodash');
+
+const string = require('../../../../../utils/string');
+
+function getTestApkPath(originalApkPath) {
+  const originalApkPathObj = path.parse(originalApkPath);
+  let tempPath = originalApkPathObj.dir.split(path.sep);
+  const splitFileName = originalApkPathObj.name.split('-');
+
+  const buildType = _.last(splitFileName);
+  const flavorDimensions = _.slice(splitFileName, 1, splitFileName.length - 1);
+
+  tempPath = _.dropRight(tempPath, 1); // buildType
+
+  let flavorDimensionsPath = '';
+  if (flavorDimensions.length > 0) {
+    flavorDimensionsPath = string.lowerCamelCaseJoin(flavorDimensions);
+    tempPath = _.dropRight(tempPath, 1); // flavorDimensions
+  }
```

Comment:
can be extracted to a nullable method IMO `getFlavorDimensionsPath(splitFileName)`, then:
```js
const flavorDimensionsPath = getFlavorDimensionsPath(..);
tempPath = flavorDimensionsPath ? _.dropRight(tempPath) : tempPath;

const finalFileName = ...

const pathComponents = [path.join(tempPath), 'androidTest', flavorDimensionsPath, buildType, finalFileName].filter(Boolean);
return path.join(pathComponents);
```

Reply:
WDYT?

Reply:
I agree!